trino/trino.go

```diff
@@ -1138,27 +1138,34 @@ func (qr *driverRows) Close() error {
 	if qr.err == sql.ErrNoRows || qr.err == io.EOF {
 		return nil
 	}
+
 	qr.err = io.EOF
-	hs := make(http.Header)
-	if qr.stmt.user != "" {
-		hs.Add(trinoUserHeader, qr.stmt.user)
-	}
-	ctx, cancel := context.WithTimeout(context.WithoutCancel(qr.ctx), DefaultCancelQueryTimeout)
-	defer cancel()
-	req, err := qr.stmt.conn.newRequest(ctx, "DELETE", qr.stmt.conn.baseURL+"/v1/query/"+url.PathEscape(qr.queryID), nil, hs)
-	if err != nil {
-		return err
-	}
-	resp, err := qr.stmt.conn.roundTrip(ctx, req)
-	if err != nil {
-		qferr, ok := err.(*ErrQueryFailed)
-		if ok && qferr.StatusCode == http.StatusNoContent {
-			qr.nextURI = ""
-			return nil
+
+	if qr.nextURI != "" {
```

Comment:
Reverse this condition and do an early return to avoid indenting the remaining code.